### PR TITLE
Fixed SRS network-policy issue for OpenShift

### PR DESCRIPTION
**Pull request by bhowd1**
- Fixed SRS network-policy issue for OpenShift (Issue #503)
- Added egress rule for openshift-dns incase of openshift cluster.
- Added new global parameter k8sProvider in backingservices helm chart to consume the kubernetes provider used to deploy the backingservice.

### CODE REVIEW

File: charts/backingservices/charts/srs/README.md [hunk 1]
```diff
@@ -57,7 +57,8 @@ To deploy Pega Platform with the SRS backing service, the SRS helm chart require
...
+| `elasticsearch`                         | Define the elasticsearch cluster configurations. The [Elasticsearch](https://github.com/helm/charts/tree/master/stable/elasticsearch/values.yaml) chart defines the values for Elasticsearch provisioning in the SRS cluster. For internally provisioned Elasticsearch the default version is set to `7.10.2`. Set the `elasticsearch.imageTag` parameter in values.yaml to `7.16.3` to use this supported version in the SRS cluster.                                                                |
+| `provider`                               | Specify your Kubernetes provider name. Supported values are [`eks`, `aks`, `minikube`, `gke`, `openshift`, `pks`]..
```
Comment (chandra-kalavakuri) [inline charts/backingservices/charts/srs/README.md:61]: please use more descriptive name for variable. `provider` -> `k8sProvider` or `kubernetesProvider` (check if we have similar variables in other charts)

Comment (bhowd1) [reply charts/backingservices/charts/srs/README.md:61]: Agreed, `k8sProvider` or `kubernetesProvider` adds more context to the variable name. 
_To answer your question:_ No we don't have `k8sProvider` or `kubernetesProvider` used in our helm charts. I can go ahead with `k8sProvider`.

**Review by chandra-kalavakuri — APPROVED** [review]

**Review by taz-pega-work — COMMENTED** [review]
this looks fine but all of the docs need to be updated with this change, now right? the 5 runbooks up in github all mention this>>
https://github.com/pegasystems/pega-helm-charts/tree/master/docs

**Comment by bhowd1** [comment]
> this looks fine but all of the docs need to be updated with this change, now right? the 5 deploy runbooks up in github all mention this parameter because they are provider-specific>> https://github.com/pegasystems/pega-helm-charts/tree/master/docs

Sure I will work on it, Thanks for your inputs @taz-pega-work !!

**Comment by bhowd1** [comment]
> this looks fine but all of the docs need to be updated with this change, now right? the 5 deploy runbooks up in github all mention this parameter because they are provider-specific>> https://github.com/pegasystems/pega-helm-charts/tree/master/docs

I updated the docs too!! @taz-pega-work @kingakowalska1 please review..

**Review by MadhuriArugula — APPROVED** [review]

**Review by taz-pega-work — APPROVED** [review]
looks great! thanks for collaborating!

**Pull request merged**